datajoint/declare.py

```diff
@@ -77,16 +90,22 @@ def compile_foreign_key(line, context, attributes, primary_key, attr_sql, foreig
     """
     # Parse and validate
     from .table import Table
+    from .query import Projection
+
+    new_style = True   # See issue #436.  Old style to be deprecated in a future release
     try:
         result = foreign_key_parser.parseString(line)
-    except pp.ParseException as err:
-        raise DataJointError('Parsing error in line "%s". %s.' % (line, err))
+    except pp.ParseException:
+        try:
+            result = foreign_key_parser_old.parseString(line)
```

Comment:
I think it would be best to throw a warning of some kind to indicate that they should update their syntax.

Reply:
Perhaps in a future release after the complete documentation is in place.  How about we make the warning in release 0.12 and raise an exception in 0.13?

Reply:
Ok I'm cool with waiting to throw a warning till 0.12. Let's see about turning it into an exception. It may just be more appropriate to make a release of 1.0.0 where we switch on exception for all warnings...